plugins/nf-amazon/src/main/nextflow/cloud/aws/nio/S3Client.java

```diff
@@ -583,7 +583,7 @@ protected void restoreFromGlacier(String bucketName, String key) {
 			RestoreObjectRequest request = new RestoreObjectRequest(bucketName, key);
 
 			String storageClass = client.getObjectMetadata(bucketName, key).getStorageClass();
-			if( storageClass != "INTELLIGENT_TIERING" )
+			if( !storageClass.equals("INTELLIGENT_TIERING") )
```

Comment:
Cannot `storageClass` be null?

Reply:
I don't think that's possible because it comes from the object metadata. I don't think an object can have an undefined storage class.

Reply:
Better being defensive. a trick could be to reverse the comparison i.e `!"INTELLIGENT_TIERING".equals(storageClass)` tho not sure the logic makes sense

Reply:
```suggestion
			if( storageClass && !storageClass.equals("INTELLIGENT_TIERING") )
```